cmd/upgrade.go

```diff
@@ -0,0 +1,209 @@
+package cmd
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/codegangsta/cli"
+	"github.com/kardianos/osext"
+)
+
+var (
+	// BuildOS is the GOOS used during the build process
+	BuildOS string
+	// Build ARMV is the GOARM used during the build process
+	BuildARM string
+	// BuildARCH is the GOARCH used during the build process
+	BuildARCH string
+)
+
+// Upgrade command allows the user to upgrade to the latest CLI version
+func Upgrade(ctx *cli.Context) {
+	client := http.Client{Timeout: 5 * time.Second}
+	rel, err := checkLatestRelease(client)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// TODO: This checks strings against string
+	// Version 2.2.0 against release 2.1.0
```

Comment:
The TODO comment is a bit unclear. Do you mean that we should check semver and only auto upgrade minor or patch versions automatically?